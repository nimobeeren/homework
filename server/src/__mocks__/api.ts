export const resolvers = {
  Query: {
    foo: () => 'bar'
  }
};
