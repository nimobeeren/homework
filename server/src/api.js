const resolvers = {
  Query: {
    videos: () => [{ title: 'Hello World', path: '/hello/world.mp4' }]
  }
};

module.exports = resolvers;
