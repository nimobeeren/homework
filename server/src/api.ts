export const resolvers = {
  Query: {
    files: () => [
      { title: 'Hello World', path: '/hello/world.mp4' },
      { title: 'Breaking Bad', path: '/hello/breaking-bad.mp4' }
    ]
  }
};
