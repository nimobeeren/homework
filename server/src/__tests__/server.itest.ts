import { ApolloServer } from 'apollo-server-koa';
import { createTestClient } from 'apollo-server-testing';
import gql from 'graphql-tag';
import typeDefs from '../../../graphql/src/schema.graphql';
import { resolvers } from '../api';
import GET_FILE_LIST from '../../../graphql/src/documents/queries.graphql';

describe('Integration: Server', () => {
  it('fetches file list', async () => {
    console.log({ GET_FILE_LIST });
    const parsed = gql`
      ${GET_FILE_LIST}
    `;
    console.log(JSON.stringify(parsed, null, 2));

    // create a test server to test against, using our production typeDefs,
    // resolvers, and dataSources.
    const server = new ApolloServer({ typeDefs, resolvers });

    // use the test server to create a query function
    const { query } = createTestClient(server);

    // run query against the server and snapshot the output
    const res = await query({ query: GET_FILE_LIST });
    expect(res).toMatchSnapshot();
  });
});
