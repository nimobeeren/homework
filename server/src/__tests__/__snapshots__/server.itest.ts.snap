// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration: Server fetches file list 1`] = `
Object {
  "data": Object {
    "files": Object {
      "items": Array [
        Object {
          "path": "/hello/world.mp4",
          "title": "Hello World",
        },
        Object {
          "path": "/hello/breaking-bad.mp4",
          "title": "Breaking Bad",
        },
      ],
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
